### PR TITLE
User rest api

### PR DESCRIPTION
**Pull request by VitaliiDenysyuk**
- [+ ] move env variables to .env

- [ +] implement "users" REST API

- [+ ] connect with mongodb

### CODE REVIEW

File: src/db.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const MongoClient = require('mongodb').MongoClient;
```
Comment (VladislavPokhyl) [inline src/db.ts:1]: It is better to use  import 'statement' then 'require'. E.g. import { MongoClient } from "mongodb";
https://flexiple.com/javascript-require-vs-import/#:~:text=One%20of%20the%20major%20differences,js%20extension%20as%20opposed%20to%20.

File: src/repositories/user.repository.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import * as db from '../db';
+const ObjectId = require('mongodb').ObjectId;
+
+export function getUsersData(done: any) {
```
Comment (VladislavPokhyl) [inline src/repositories/user.repository.ts:4]: For such resource-intensive processes as accessing the database, you need to use the async-await construct.

File: src/db.ts (hunk 1 above)
Comment (VitaliiDenysyuk) [reply src/db.ts]: > It is better to use import 'statement' then 'require'. E.g. import { MongoClient } from "mongodb"; https://flexiple.com/javascript-require-vs-import/#:~:text=One%20of%20the%20major%20differences,js%20extension%20as%20opposed%20to%20.

corrected

**Comment by VitaliiDenysyuk** [comment]
corrected

File: src/app.ts [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
 
+import logger from 'morgan';
+
+
+require('dotenv').config();
 var indexRouter = require('./routes/index');
 var usersRouter = require('./routes/users');
 
-var app = express();
+export var app = express();
```
Comment (alexandrtovmach) [inline src/app.ts:12]: do we need this export here?

File: src/db.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { MongoClient } from 'mongodb';
```
Comment (alexandrtovmach) [inline src/db.ts:1]: better to use mongoose

File: src/repositories/user.repository.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
+import * as db from '../db';
+
+import { ObjectId } from 'mongodb';
+
+export function getUsersData(done: any) {
+    db.get().collection('users').find().toArray((err: any, docs: any) => {
```
Comment (alexandrtovmach) [inline src/repositories/user.repository.ts:6]: don't use callbacks

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
- use mongoose instead of pure mongo client
- change flow to async/await

File: src/app.ts (hunk 3 above)
Comment (VitaliiDenysyuk) [reply src/app.ts:12]: yes. We import app variable from this module in index.ts

File: src/db.ts (hunk 4 above)
Comment (VitaliiDenysyuk) [reply src/db.ts]: corrected

File: src/repositories/user.repository.ts (hunk 5 above)
Comment (VitaliiDenysyuk) [reply src/repositories/user.repository.ts]: corrected

**Comment by VitaliiDenysyuk** [comment]
> * use mongoose instead of pure mongo client
> * change flow to async/await

ready

File: src/routes/users.ts [hunk 6]
```diff
@@ -1,9 +1,68 @@
...
+import { isAuthorised } from '../middlewares/auth.middleware';
+import {
+  getListUsers,
+  addUser,
+  findUserById,
+  changeUserById,
+  deleteUserById,
+} from '../services/user.service';
+
+export const router = express.Router();
```
Comment (alexandrtovmach) [inline src/routes/users.ts:11]:
```suggestion
const router = express.Router();
```

File: src/routes/users.ts [hunk 7]
```diff
@@ -1,9 +1,68 @@
...
+  deleteUserById(req.params.id)
+    .then((result) => {
+      res.send(result)
+    }).catch((error) => {
+      console.log(error);
+      res.status(400).send(error);
+    })
+});
+
+
```
Comment (alexandrtovmach) [inline src/routes/users.ts:68]: export default router;

File: src/app.ts [hunk 8]
```diff
@@ -1,12 +1,21 @@
...
-var indexRouter = require('./routes/index');
-var usersRouter = require('./routes/users');
 
-var app = express();
+import dotenv from 'dotenv';
+dotenv.config();
+
+import {
+    router as indexRouter
+} from './routes/index';
```
Comment (alexandrtovmach) [inline src/app.ts:12]: import indexRouter from './routes/index';

File: src/repositories/user.repository.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        let userInDatabase: any;
+        await findUserByIdData(id)
+            .then((result) => {
+                if (!result) {
+                    throw `Error: User by id ${id} is not found!`;
+                }
+                userInDatabase = result;
+            }).catch((err) => {
+                throw err;
+            });
```
Comment (alexandrtovmach) [inline src/repositories/user.repository.ts:36]:
```
const userData = await findUserByIdData(id);
if (!userData) {
    throw `Error: User by id ${id} is not found!`;
}
```

File: src/services/user.service.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    deleteUserByIdData,
+} from '../repositories/user.repository';
+
+export const getListUsers = async () => {
+    return getUsersData();
+}
+
+export const addUser = async (user: any) => {
+    if (user) {
+        return await addUserData(user)
```
Comment (alexandrtovmach) [inline src/services/user.service.ts:15]: return addUserData(user);

**Comment by VitaliiDenysyuk** [comment]
all comments are taken into account

**Review by alexandrtovmach — APPROVED** [review]
lgtm

**Pull request merged**